koku/api/report/view.py

```diff
@@ -179,14 +179,14 @@ def costs(request):
 @api_view(http_method_names=['GET'])
 @authentication_classes([SessionAuthentication, TokenAuthentication])
 @permission_classes([IsAuthenticated])
-def inventory(request):
+def instance_type(request):
     """Get inventory data.
 
-    @api {get} /api/v1/reports/inventory/ Get inventory data
-    @apiName getInventoryData
+    @api {get} /api/v1/reports/inventory/instance-type/ Get inventory instance type data
```

Comment:
It looks like you're reverting a previous API change where we decided to _not_ do `/instance` and `/storage` endpoints because that won't scale and is EC2-centric without acknowledging the full breadth of AWS services.

Reply:
Yep. I decided to focus on the current dashboard cards.

I mentioned during our stand up I was having trouble with the `/inventory/` API as I don't understand what it returns if you do a *GET* request without any filtering against it (i.e. no query parameters). Since the resources are all different units of usage there is no way to do an aggregation.  To me it feels wrong to return an error to say a query parameter is required. Maybe the answer is there should be a query parameter that controls aggregation vs. returning line items? Do have the same interaction for the cost endpoint? Is there any difference in output if the aggregation parameter isn't given?

So instead of solving the problem last week, I deferred the API decision and created the current endpoint specific PR (with the caveat that it will change as we work through the API for UI/QE).

Reply:
I have some ideas on those questions, but that discussion is probably outside the scope of this PR.

I can live with delaying those decisions for now. :smile: